File: 240128/마라톤 중간에 택시타기 2/taking-a-taxi-in-the-middle-of-the-marathon-2.py
import sys
n=int(input())

coord=[list(map(int,input().split())) for _ in range(n)]
ans=sys.maxsize
for i in range(1,n-1):
    dist=0
    for j in range(n-1):
        if j+1==i:
            dist+=abs(coord[j][0]-coord[j+2][0])+abs(coord[j][1]-coord[j+2][1])
        elif j==i:
            continue 
        else:
            dist+=abs(coord[j][0]-coord[j+1][0])+abs(coord[j][1]-coord[j+1][1])
        #print(i,dist)
    ans=min(ans,dist)
    #print(ans)

print(ans)

File: 260106/마라톤 중간에 택시타기 2/taking-a-taxi-in-the-middle-of-the-marathon-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const int_max = Number.MAX_SAFE_INTEGER;
const n = Number(input[0]);
const arr = Array.from({length: n}, (_, i) => input[i + 1].split(' ').map(Number));

// Please Write your code here.
function calculateDistance(x1,y1,x2,y2){
    return Math.abs(x1-x2) + Math.abs(y1-y2);
}

let ans = int_max;

for(let i=1; i<n-1; i++){

    let curArr = [...arr];

    curArr.splice(i,1);

    let curDistance = 0;

    for(let j=0; j<curArr.length-1; j++){   
        const [x1, y1] = curArr[j];
        const [x2, y2] = curArr[j+1];

        curDistance += calculateDistance(x1, y1, x2, y2);
    }

    ans = Math.min(ans, curDistance);
}

console.log(ans)